src/webgpu/api/validation/createRenderPipeline.spec.ts

```diff
@@ -0,0 +1,226 @@
+export const description = `
+createRenderPipeline validation tests.
+`;
+
+import { poptions } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { kAllTextureFormats, kAllTextureFormatInfo } from '../../capability_info.js';
+
+import { ValidationTest } from './validation_test.js';
+
+class F extends ValidationTest {
+  getDescriptor(
+    options: {
+      primitiveTopology?: GPUPrimitiveTopology;
+      colorStates?: GPUColorStateDescriptor[];
+      sampleCount?: number;
+      depthStencilState?: GPUDepthStencilStateDescriptor;
+    } = {}
+  ): GPURenderPipelineDescriptor {
+    const defaultColorStates: GPUColorStateDescriptor[] = [{ format: 'rgba8unorm' }];
+    const {
+      primitiveTopology = 'triangle-list',
+      colorStates = defaultColorStates,
+      sampleCount = 1,
+      depthStencilState,
+    } = options;
+
+    const format = colorStates.length ? colorStates[0].format : 'rgba8unorm';
+
+    let fragColorType;
+    if (format.endsWith('sint')) {
+      fragColorType = 'i32';
+    } else if (format.endsWith('uint')) {
+      fragColorType = 'u32';
+    } else {
+      fragColorType = 'f32';
+    }
+
+    return {
+      vertexStage: {
+        module: this.device.createShaderModule({
+          code: `
+            [[builtin(position)]] var<out> Position : vec4<f32>;
+
+            [[stage(vertex)]] fn main() -> void {
+              Position = vec4<f32>(0.0, 0.0, 0.0, 1.0);
+              return;
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragmentStage: {
+        module: this.device.createShaderModule({
+          code: `
+            [[location(0)]] var<out> fragColor : vec4<${fragColorType}>;
+            [[stage(fragment)]] fn main() -> void {
+              fragColor = vec4<${fragColorType}>(vec4<f32>(0.0, 1.0, 0.0, 1.0));
```

Comment:
maybe it will be better to choose the suffix above which is either ".0", "u", or ""